Framework/DataHandling/src/LoadSampleShape.cpp

```diff
@@ -140,6 +140,123 @@ std::unique_ptr<Geometry::MeshObject> readSTLSolid(std::ifstream &file,
   return nullptr;
 }
 
+bool getOFFline(std::ifstream &file, std::string &line) {
+  // Get line from OFF file ignoring blank lines and comments
+  // The line output is ready trimmed
+  if (!getline(file, line)) {
+    return false;
+  }
+  boost::trim(line);
+  while (line.empty() || line.substr(0, 1) == "#") {
+    if (!getline(file, line)) {
+      return false;
+    }
+    boost::trim(line);
+  }
+  return true;
+}
+
+void readOFFVertices(std::ifstream &file, uint16_t nV,
+                     std::vector<V3D> &vertices) {
+  std::string line;
+  V3D vertex;
+  for (uint16_t i = 0; i < nV; i++) {
+    if (getOFFline(file, line)) {
+      std::vector<std::string> tokens;
+      boost::split(tokens, line, boost::is_any_of(" "),
+                   boost::token_compress_on);
+      if (tokens.size() == 3) {
+        vertex.setX(boost::lexical_cast<double>(tokens[0]));
```

Comment:
could probably just emplace into `vertices` here without using `vertex`. You'll save on all the copies for large numbers of vertices.
```
vertices.emplace_back(boost::lexical_cast<double>(tokens[0]), //x
                      boost::lexical_cast<double>(tokens[1]), //y
                      boost::lexical_cast<double>(tokens[2])) //z
````